### ISSUE TITLE
Disable "short" DB names in replication requests

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#1816]
Since 2.0.0 has been released, there have been a number of issues with specifying "short" / "local" database names in replication requests. The most recent encounter with problems concerning this syntax is:

https://github.com/apache/couchdb/issues/1550#issuecomment-448149590

We need to do one of two things:
1. Fix "local" replication endpoints completely. My understanding from @nickva is that this is very difficult to accomplish, likely impossible.
2. Remove the ability to specify replication `source` and `target` as just a database name, both in the HTTP API as well as in Fauxton.

In short, it's not right for us to continue to encourage use of an API endpoint that is knowingly broken, and ignore the problem for 3 major releases.

**Comment by nickva** [comment]
+1 on "Remove the ability to specify replication source and target as just a database name, both in the HTTP API as well as in Fauxton" point

To expand on it:

Since 2.x local replication endpoints refer only to node-local databases. In 1.x they were "proper" databases so this was a useful thing to have. Now, it is mostly confusing, as these refer to databases created on the :5986 port, which is very surprising and mostly useless. A user who migrated from 1.x might be very confused as to what's happening and why things don't work as expected anymore.

Moreover, replication jobs are distributed to individual cluster nodes in an consistent but essentially opaque manner from a user's point of view.  So unless a cluster has only one node,  a replication job might end up on node 2, or node 3 for example, and would be pick the local endpoint dbs on that node only.

So perhaps we could disable it by default. Then users  would toggle a config value say `allow_node_local_endpoints = true` to use it. Eventually in a future release, it goes away altogether and we get to clean the complicated replicator code that handles both local and remote endpoints. Or just warn the disable it altogether, without bothering with a config toggle.

**Comment by nickva** [comment]
Fixed in https://github.com/apache/couchdb/pull/2089
